plugin-api/src/main/java/org/hyperledger/besu/plugin/services/securitymodule/SecurityModuleProvider.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.plugin.services.securitymodule;
+
+import org.hyperledger.besu.plugin.Unstable;
+import org.hyperledger.besu.plugin.services.BesuConfiguration;
+
+import java.util.function.Function;
+
+/**
+ * Represents Security Module Provider Function which takes an instance of BesuConfiguration and
+ * produces an instance of SecurityModule.
+ */
+@Unstable
+@FunctionalInterface
+public interface SecurityModuleProvider extends Function<BesuConfiguration, SecurityModule> {}
```

Comment:
Still prefer to not have this extend Function. Think this would be much cleaner with a just a SecurityModule create(BesuConfiguration config) method.

Reply:
>Think this would be much cleaner with a just a SecurityModule create(BesuConfiguration config) method.

It would also be more consistent with the approach use in the other plugins.

Reply:
👍